src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala

```diff
@@ -73,7 +73,7 @@ class Row private[util] (private val headerIndices: Map[String,Int], private val
   /** Gets a value from the specified column. If the value is null or empty returns None.  If
     * `allowMissingColumn` is true, then None is returned if the given column name is not present. */
   def get[A](column: String, allowMissingColumn: Boolean = false)(implicit tag: ru.TypeTag[A]): Option[A] =
-    if (allowMissingColumn) headerIndices.get(column).map(apply[A]) else get(headerIndices(column))
+    if (allowMissingColumn && ! headerIndices.contains(column)) None else get(headerIndices(column))
```

Comment:
How about this? If you go with your implementation, please change `! headerIndices` to `!headerIndices`
```suggestion
    if (allowMissingColumn) headerIndices.get(column).flatMap(get[A]) else get(headerIndices(column))
```